app/src/main/java/com/cornellappdev/transit/ui/viewmodels/ArriveByUIState.kt

```diff
@@ -0,0 +1,92 @@
+package com.cornellappdev.transit.ui.viewmodels
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.cornellappdev.transit.util.TimeUtils
+import java.time.Instant
+import java.util.Date
+
+/**
+ * Wrapper for state of the search bar in HomeScreen
+ */
+sealed class ArriveByUIState {
+    data class LeaveNow(val tag: String = "Leave now") :
+        ArriveByUIState() {
+        val date
+            @RequiresApi(Build.VERSION_CODES.O)
+            get() = Date.from(Instant.now())
+
+    }
+
+    data class LeaveAt(
+        val date: Date,
+        val tag: String = "Leave at",
+    ) :
+        ArriveByUIState()
+
+    data class ArriveBy(
+        val date: Date,
+        val tag: String = "Arrive by"
+    ) :
+        ArriveByUIState()
+}
```

Comment:
I immediately suspect we can add inheritance here to improve type checking! Right now, I assume whenever you want to access a `data` or `tag`, you have to first cast it, then pull the date / tag. With this:

```kotlin
sealed class ArriveByUIState {
    abstract val date: Date
    abstract val tag: String

    data class LeaveNow @RequiresApi(Build.VERSION_CODES.O) constructor(
        override val tag: String = "Leave now"
    ) : ArriveByUIState() {
        override val date: Date
            @RequiresApi(Build.VERSION_CODES.O)
            get() = Date.from(Instant.now())
    }

    data class LeaveAt(
        override val date: Date,
        override val tag: String = "Leave at"
    ) : ArriveByUIState()

    data class ArriveBy(
        override val date: Date,
        override val tag: String = "Arrive by"
    ) : ArriveByUIState()
}
```

You should be able to pull them out without having to type cast first!